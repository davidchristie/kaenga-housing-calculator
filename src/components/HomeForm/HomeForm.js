import React, { Component, PropTypes } from 'react'
import { Panel } from 'react-bootstrap'
import { Field } from 'redux-form'

import BathtubsField from './BathtubsField'
import DoubleBedroomsField from './DoubleBedroomsField'
import SingleBedroomsField from './SingleBedroomsField'
import ShowersField from './ShowersField'
import StudyField from './StudyField'
import ToiletsField from './ToiletsField'

class HomeForm extends Component {
  render () {
    const { handleSubmit } = this.props
    return (
      <form onSubmit={handleSubmit}>
        <Panel header='Home'>
          <Field component={DoubleBedroomsField} name='doubleBedrooms' />
          <Field component={SingleBedroomsField} name='singleBedrooms' />
          <Field component={StudyField} name='study' />
          <Field component={BathtubsField} name='bathtubs' />
          <Field component={ShowersField} name='showers' />
          <Field component={ToiletsField} name='toilets' />
        </Panel>
      </form>
    )
  }
}

HomeForm.propTypes = {
  handleSubmit: PropTypes.func
}

export default HomeForm
