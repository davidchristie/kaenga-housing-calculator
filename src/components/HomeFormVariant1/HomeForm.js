import React, { Component, PropTypes } from 'react'
import { Panel, Well } from 'react-bootstrap'
import { FieldArray, reduxForm } from 'redux-form'

import ItemListControl from '../controls/ItemListControl'
import BathroomFields from './BathroomFields'
import BedroomFields from './BedroomFields'

class HomeForm extends Component {
  render () {
    const { handleSubmit } = this.props
    return (
      <form onSubmit={handleSubmit}>
        <Panel bsStyle='warning' header='Home (Variant #1)'>
          <Well>
            <h4>Bedrooms</h4>
            <FieldArray
              component={ItemListControl}
              defaultItemValues={{size: 'double'}}
              itemComponent={BedroomFields}
              name='bedrooms'
            />
          </Well>
          <Well>
            <h4>Bathrooms</h4>
            <FieldArray
              component={ItemListControl}
              defaultItemValues={{
                bath: false,
                shower: false,
                toilet: true
              }}
              itemComponent={BathroomFields}
              name='bathrooms'
            />
          </Well>
        </Panel>
      </form>
    )
  }
}

HomeForm.propTypes = {
  handleSubmit: PropTypes.func
}

export default reduxForm({
  form: 'homeVariant1'
})(HomeForm)
