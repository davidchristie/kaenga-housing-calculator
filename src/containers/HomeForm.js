import { Field, reduxForm } from 'redux-form'

import React, { Component } from 'react'
import { Panel } from 'react-bootstrap'

import Checkbox from '../components/controls/Checkbox'
import NumberInput from '../components/controls/NumberInput'
import Anchor from '../components/layout/Anchor'
import Tip from '../components/misc/Tip'

class HomeForm extends Component {
  render () {
    const { handleSubmit } = this.props
    return (
      <form onSubmit={handleSubmit}>
        <Panel header='Home Builder'>
          <Anchor id='home-builder' />
          <h3>Home Design</h3>
          <Field
            component={NumberInput}
            label='Number of occupants'
            name='occupants'
          />
          <Field
            component={NumberInput}
            label='Number of bedrooms'
            name='bedrooms'
          />
          <Field
            component={NumberInput}
            label='Bathrooms'
            name='bathrooms'
          />
          <Field
            component={Checkbox}
            label='Shared guest studio'
            name='study'
          />
          <Tip>
            Sharing a guest studio with others is significantly cheaper than having your own guest room. If you decide to do this, make sure to reduce the number of bedrooms you need above
          </Tip>
          <Panel
            bsStyle='warning'
            header='Old fields'
            style={{display: 'none'}}
          >
            <Field
              component={NumberInput}
              label='How many double bedrooms do you need?'
              name='doubleBedrooms'
            />
            <Field
              component={NumberInput}
              label='How many single bedrooms do you need?'
              name='singleBedrooms'
            />
            <Field
              component={NumberInput}
              label='How many bathrooms with a bathtub do you need?'
              name='bathtubs'
            />
            <Field
              component={NumberInput}
              label='How many bathrooms with a shower do you need?'
              name='showers'
            />
            <Field
              component={NumberInput}
              label='How many stand alone toilets do you need?'
              name='toilets'
            />
          </Panel>
        </Panel>
      </form>
    )
  }
}

export default reduxForm({
  destroyOnUnmount: false,
  form: 'homeBuilder'
})(HomeForm)
