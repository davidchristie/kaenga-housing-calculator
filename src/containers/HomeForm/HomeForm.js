import { reduxForm } from 'redux-form'

import HomeForm from '../../components/HomeForm'

export default reduxForm({
  form: 'home'
})(HomeForm)
