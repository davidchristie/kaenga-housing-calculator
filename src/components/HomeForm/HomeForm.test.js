import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import HomeForm from './HomeForm'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<HomeForm />)).toMatchSnapshot()
})
