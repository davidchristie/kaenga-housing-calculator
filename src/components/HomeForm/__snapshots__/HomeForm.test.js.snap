exports[`test matches snapshot 1`] = `
<form
  onSubmit={undefined}>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}
    header="Home">
    <Field
      component={[Function]}
      name="doubleBedrooms" />
    <Field
      component={[Function]}
      name="singleBedrooms" />
    <Field
      component={[Function]}
      name="study" />
    <Field
      component={[Function]}
      name="bathtubs" />
    <Field
      component={[Function]}
      name="showers" />
    <Field
      component={[Function]}
      name="toilets" />
  </Panel>
</form>
`;
